delfin/api/v1/storages.py

```diff
@@ -83,6 +83,8 @@ def create(self, req, body):
         lock_name = 'storage-create-' + host
         lock = coordination.Lock(lock_name)
 
+        if self._storage_exist(ctxt, access_info_dict):
+            raise exception.StorageAlreadyExists()
         with lock:
             if self._storage_exist(ctxt, access_info_dict):
```

Comment:
This is suplicate of line 86?

Reply:
Not duplicated. Line 86 and line 89 are for different scenario:
Line 86: storage has been registered, so line 87 would raise an exception
Line 89: request 1 and request 2 for same access info which has not been registered. Because the storage has not been registered, so for both request, line 87 would not raise exception. And came to line 88, one request could get the lock, another would wait for lock. The request which get the lock at first would registered successfully, then release the lock, then another request would get the lock and come to line 89, and it would raise the exception.